www/js/controllers.js

```diff
@@ -156,32 +180,75 @@ angular.module('starter.controllers', [])
 
   // where the map is initiated and called
   navigator.geolocation.getCurrentPosition($scope.drawMap);
-  navigator.geolocation.getCurrentPosition(function(position){
-    $scope.coordinates = {
-      lat: position.coords.latitude, 
-      lng: position.coords.longitude
-    };
-  });
-
-  $scope.sendCurrentLocation = function(){
-    console.log($scope.coordinates);
-    var geocoder = new google.maps.Geocoder;
-    geocoder.geocode({'location': $scope.coordinates}, function(results, status){
-      console.log(results);
-    });
-    // send out http put request to location
+
+  $scope.sendCenterLocation = function(){
+    if(!$scope.center_coords){
+      console.log('My current location is X:' + $scope.myLocation.lat + ', Y: ' + $scope.myLocation.lng);
+
+      sendPickupRequest($scope.myLocation.lat, $scope.myLocation.lng);
+
+    } else{
+      console.log('I selected location X:' + $scope.center_coords.G + ', Y:' + $scope.center_coords.K);
+
+      sendPickupRequest($scope.center_coords.G, $scope.center_coords.K);
+
+    }
   }
 
-  $scope.sendPickupLocation = function(query){
-    // console.log(query);
-    var geocoder = new google.maps.Geocoder;
-    geocoder.geocode({'address': query + ', Hong Kong'}, function(results, status){
-      console.log('query----->', results[0].formatted_address, 'coords----->' ,results[0].geometry.location);
-      $scope.query_coords = [results[0].geometry.location.G, results[0].geometry.location.K];
-      console.log($scope.query_coords);
+  function sendPickupRequest(lat, lng){
+    var request = {request: {
+      latitude: lat,
+      longitude: lng
+    }};
+
+    $ionicLoading.show({
+      templateUrl: 'templates/notification/waiting_request_page.html',
     });
+
+    $http.post('http://vala-api.herokuapp.com/api/v1/users/'+ $scope.currentUser.id+'/requests', request).then(function(response){
+      console.log(response);
+      $scope.requestMade = true;
+      $scope.LastRequestId = response.data.id;
+      console.log($scope.LastRequestId);
+    }).catch(function(response){
+      console.log(response);
+      $ionicLoading.hide();
+      $ionicPopup.alert({
+        title: 'There was an error in your request.',
+        template: 'Please try again later.'
+      });
+
+    })
+  }
+
+  $scope.cancelRequest = function(){
+    console.log('cancelRequest');
+    $http.put('http://vala-api.herokuapp.com/api/v1/requests/'+ $scope.LastRequestId).then(function(response){
```

Comment:
@f3r This is good for "refactoring". I will do this by the next PR. Is that okay?